### PR TITLE
[TSE] Implement Exchange Operation Service

### LINKED ISSUE #8: [TSE] Implement Exchange Operation Service
### Description
This service will be responsible for persisting every exchange operation in the database.

#### Puropose
Thanks to this, we will be able to recover from failures, gain insights into exchange activity, and avoid relying on exchange API calls for internal operations when fetching order data.

### To Do
- Implement `saveExchangeOperation` method
- Implement `saveOrderData` method

### Deliverables
- [ ] All tasks completed
- [ ] Unit tests
- [ ] Documentation

### PR DESCRIPTION
**Pull request by MWBlocky**
## Description

This Pull Request introduces a comprehensive service for persisting exchange operations in a database. The service is designed to ensure that every interaction with supported exchanges is reliably stored, enabling recovery from failures and providing insights into exchange activities without relying on direct API calls for internal operations.

## Summary of changes

- **Implemented `saveExchangeOperation` method**: Stores details of exchange operations, such as order placements, cancellations in the database.
- **Implemented `saveOrderData` method**: Persists order data including status, price, and quantity to the database, ensuring accurate records for future reference.
- **Added unit and e2e tests**: Comprehensive tests were added to validate the functionality of each method and to handle edge cases.

## Related issues

Fixes #8

### CODE REVIEW

File: packages/tse-backend/src/modules/exchange-operation/exchange-operation.repository.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+@Injectable()
+export class ExchangeOperationRepository {
+  constructor(
+    @InjectRepository(Order)
+    private readonly orderRepository: Repository<Order>,
+    @InjectRepository(Operation)
+    private readonly operationRepository: Repository<Operation>,
+  ) {}
+
```
Comment (piotrswierzy) [inline packages/tse-backend/src/modules/exchange-operation/exchange-operation.repository.ts:15]: For each entity have a separate repository and service to handle them

File: packages/tse-backend/src/modules/exchange-operation/exchange-operation.service.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  ExchangeOperationCommand,
+  OrderCommand,
+} from './model/exchange-operation.model';
+import { CustomLogger } from '../logger/logger.service';
+import { ExchangeOperationRepository } from './exchange-operation.repository';
+import { OrderStatus } from '../../common/enums/exchange-operation.enums';
+import { Order } from '../../common/entities/order.entity';
+
+@Injectable()
+export class ExchangeOperationService {
```
Comment (piotrswierzy) [inline packages/tse-backend/src/modules/exchange-operation/exchange-operation.service.ts:13]: Create seperate service for ExchangeOrder and ExchangeOperation

**Review by piotrswierzy — CHANGES_REQUESTED** [review]

File: packages/tse-backend/src/common/interfaces/exchange-operation.interfaces.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { Order } from '../entities/order.entity';
+import { OrderStatus } from '../enums/exchange-operation.enums';
+
+export interface SaveOperationDto {
+  status: OrderStatus;
+  details: Record<string, any>;
+  order: Order;
+}
+
+export interface PersistOperationDto {
```
Comment (piotrswierzy) [inline packages/tse-backend/src/common/interfaces/exchange-operation.interfaces.ts:10]: What's the difference in `Persist` vs `Save`?

**Review by piotrswierzy — CHANGES_REQUESTED** [review]

File: packages/tse-backend/src/common/interfaces/exchange-operation.interfaces.ts (hunk 3 above)
Comment (MWBlocky) [reply packages/tse-backend/src/common/interfaces/exchange-operation.interfaces.ts:10]: `Persist` is used in the service, while `Save` is used in the repository.
```
  async persistOperationData(order: Order, data: PersistOperationDto) {
    await this.repository.save({
      status: data.status,
      details: data.details,
      order,
    } as SaveOperationDto);
  }
```

**Review by piotrswierzy — APPROVED** [review]
